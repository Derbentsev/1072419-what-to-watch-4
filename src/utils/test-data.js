import {FilterNames} from '@consts/';


const films = [
  {
    id: 1,
    title: `title1`,
    poster: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    genre: `genre1`,
    dateRelease: 2011,
    cover: `the-grand-budapest-hotel-poster.jpg`,
    videoSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    previewVideoSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    director: `director1`,
    actors: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    ratingScore: 1,
    ratingCount: 240,
    description: `description1`,
    runTime: 99,
    reviews: [
      {
        comment: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
        rating: 8.9,
        author: `Kate Muir`,
        date: `December 24, 2016`,
      },
    ]
  },
  {
    id: 2,
    title: `title1`,
    poster: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    genre: `genre1`,
    dateRelease: 2011,
    cover: `the-grand-budapest-hotel-poster.jpg`,
    videoSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    previewVideoSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    director: `director1`,
    actors: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    ratingScore: 1,
    ratingCount: 240,
    description: `description1`,
    runTime: 99,
    reviews: [
      {
        comment: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
        rating: 8.9,
        author: `Kate Muir`,
        date: `December 24, 2016`,
      },
    ]
  },
  {
    id: 3,
    title: `title1`,
    poster: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    genre: `genre1`,
    dateRelease: 2011,
    cover: `the-grand-budapest-hotel-poster.jpg`,
    videoSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    previewVideoSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    director: `director1`,
    actors: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    ratingScore: 1,
    ratingCount: 240,
    description: `description1`,
    runTime: 99,
    reviews: [
      {
        comment: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
        rating: 8.9,
        author: `Kate Muir`,
        date: `December 24, 2016`,
      },
    ]
  },
];

const filmPromo = {
  id: 1,
  title: `title1`,
  poster: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  genre: `genre1`,
  dateRelease: 2011,
  cover: `the-grand-budapest-hotel-poster.jpg`,
  videoSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  previewVideoSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  director: `director1`,
  actors: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
  ratingScore: 1,
  ratingCount: 24,
  description: `description1`,
  runTime: 103,
  reviews: [
    {
      comment: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
      rating: 8.3,
      author: `Kate Muir`,
      date: `December 24, 2016`,
    },
  ]
};

const activeFilm = {
  id: 2,
  title: `title1`,
  poster: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  genre: `genre1`,
  dateRelease: 2011,
  cover: `the-grand-budapest-hotel-poster.jpg`,
  videoSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  previewVideoSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  director: `director1`,
  actors: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
  ratingScore: 1,
  ratingCount: 24,
  description: `description1`,
  runTime: 103,
  reviews: [
    {
      comment: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
      rating: 8.3,
      author: `Kate Muir`,
      date: `December 24, 2016`,
    },
  ]
};

const mockFilmsStore = {
  films,
  filmPromo,
  activeFilm,
};

const mockPlayerStore = {
  activeFullVideoPlayer: null,
};

const mockFilterStore = {
  currentFilter: FilterNames.ALL,
};

const mockUserStore = {
  authorizationStatus: `NO_AUTH`,
  authorizationError: ``,
};

export {
  mockFilmsStore,
  mockPlayerStore,
  mockFilterStore,
  mockUserStore,
};
