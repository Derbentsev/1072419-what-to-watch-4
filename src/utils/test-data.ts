const films = [
  {
    id: 1,
    title: `title1`,
    poster: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    genre: `genre1`,
    dateRelease: 2011,
    isFavorite: true,
    cover: `the-grand-budapest-hotel-poster.jpg`,
    videoSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    previewVideoSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    backgroundImage: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    backgroundColor: `green`,
    director: `director1`,
    actors: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    ratingScore: 1,
    ratingCount: 240,
    description: `description1`,
    runTime: 99,
  },
  {
    id: 2,
    title: `title1`,
    poster: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    genre: `genre1`,
    dateRelease: 2011,
    isFavorite: true,
    cover: `the-grand-budapest-hotel-poster.jpg`,
    videoSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    previewVideoSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    backgroundImage: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    backgroundColor: `green`,
    director: `director1`,
    actors: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    ratingScore: 1,
    ratingCount: 240,
    description: `description1`,
    runTime: 99,
  },
  {
    id: 3,
    title: `title1`,
    poster: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    genre: `genre1`,
    dateRelease: 2011,
    isFavorite: false,
    cover: `the-grand-budapest-hotel-poster.jpg`,
    videoSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    previewVideoSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    backgroundImage: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    backgroundColor: `green`,
    director: `director1`,
    actors: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    ratingScore: 1,
    ratingCount: 240,
    description: `description1`,
    runTime: 99,
  },
];

const rawFilms = [
  {
    id: 1,
    name: `title1`,
    background_image: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    genre: `genre1`,
    released: 2011,
    is_favorite: true,
    poster_image: `the-grand-budapest-hotel-poster.jpg`,
    video_link: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    preview_video_link: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    background_color: `green`,
    director: `director1`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    rating: 1,
    scores_count: 240,
    description: `description1`,
    run_time: 99,
  },
  {
    id: 2,
    name: `title1`,
    background_image: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    genre: `genre1`,
    released: 2011,
    is_favorite: true,
    poster_image: `the-grand-budapest-hotel-poster.jpg`,
    video_link: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    preview_video_link: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    background_color: `green`,
    director: `director1`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    rating: 1,
    scores_count: 240,
    description: `description1`,
    run_time: 99,
  },
  {
    id: 3,
    name: `title1`,
    background_image: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    genre: `genre1`,
    released: 2011,
    is_favorite: false,
    poster_image: `the-grand-budapest-hotel-poster.jpg`,
    video_link: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    preview_video_link: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    background_color: `green`,
    director: `director1`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    rating: 1,
    scores_count: 240,
    description: `description1`,
    run_time: 99,
  }
];

const reviews = [
  {
    comment: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    rating: 8.9,
    author: `Kate Muir`,
    date: `December 24, 2016`,
  },
  {
    comment: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    rating: 8.9,
    author: `Kate Muir`,
    date: `December 24, 2016`,
  },
]

const filmPromo = {
  id: 10,
  title: `title1`,
  poster: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  genre: `genre1`,
  dateRelease: 2011,
  isFavorite: false,
  cover: `the-grand-budapest-hotel-poster.jpg`,
  videoSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  previewVideoSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  backgroundImage: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  backgroundColor: `green`,
  director: `director1`,
  actors: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
  ratingScore: 1,
  ratingCount: 240,
  description: `description1`,
  runTime: 99,
};

const mockFilmsStore = {
  films,
  filmPromo,
  favoriteFilms: films,
};

const mockFilterStore = {
  currentFilter: `All genres`,
};

const mockUserStore = {
  authorizationStatus: `NO_AUTH`,
  authorizationError: ``,
};

const mockReviewsStore = {
  reviews,
  pushReviewStatus: ``,
};


export {
  mockFilmsStore,
  mockFilterStore,
  mockUserStore,
  mockReviewsStore,
  films,
  reviews,
  rawFilms,
};
