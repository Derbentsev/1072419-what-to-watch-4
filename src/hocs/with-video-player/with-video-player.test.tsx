import withVideoPlayer from './with-video-player';
import {films} from '@utils/test-data';


interface MockComponentProps {
  children: React.ReactNode;
}

const MockComponent = (props: MockComponentProps) => {
  const {children} = props;

  return (
    <div>
      {children}
    </div>
  );
};

const MockWrappedComponent = withVideoPlayer(MockComponent);

it(`Render Movie Page`, () => {
  const tree = renderer
    .create(
        <MockWrappedComponent
          film = {films[0]}
        />
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
