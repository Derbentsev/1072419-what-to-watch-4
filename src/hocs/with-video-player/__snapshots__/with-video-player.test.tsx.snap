// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Movie Page 1`] = `
<div>
  <video
    height="175"
    muted={true}
    poster="fantastic-beasts-the-crimes-of-grindelwald.jpg"
    src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
    width="280"
  />
</div>
`;
